3.0/vector_tile.proto

```diff
@@ -44,6 +44,49 @@ message Tile {
+                
+                // Additional tags (or all the tags) of this feature may be
+                // encoded as repeated pairs of 32-bit integers, to take
+                // advantage of inline encoding of small values and of
+                // improved compression from use of repeated values.
+                //
+                // This message may only be used if the layer version is >= 3.
+                //
+                // The properties field is much like the tags value in that it is a pair of
+                // integers that reference key and value pairs. However, the value reference
+                // is a "complex_value" that combines a type and an index.
+                // 
+                // The "key_index" is much like the key index in the use for tags, but instead
+                // of pointing to "keys" field in the Layers, it points to the "string_values".
+                // This is the same value store as strings for use in values, so duplicates here
+                // will be pointing to the same indexed position. 
+                //
+                // An complex value has two parts: the lowest 4 bits are the type bits,
+                // and the remaining bits are the parameter bits. What is stored in the parameter
+                // bits is dependent on the contents of the type bits. For example for inline values,
+                // the parameter field is not an index but simply a value. For other types it is
+                // an index position into a value storage of the layer.
+                // 
+                // uint64t type = complex_value & 0x0F; // First 4 Bits
+                // uint64t parameter = complex_value >> 4;
+                // 
+                //     Type     | Id  | Parameter
+                // ---------------------------------
+                // inline sint  |  0  | value of zigzag-encoded integer (values between -2^59 to 2^59-1)
+                // inline uint  |  1  | value of unsigned integer (values between 0 to 2^60-1)
+                // bool/null    |  2  | value of 0 = false, 1 = true, 2 = null
+                // float        |  3  | index to float_values in layer
+                // double       |  4  | index to double_values in layer
+                // string       |  5  | index to string_values in layer
+                // int          |  6  | index to sfixed64_values in layer
+                // uint         |  7  | index to fixed64_values in layer
+                // list / map   |  8  | (if 4th bit is 0 is list)
+                //              |     |   remaining bits are length of the list where
+                //              |     |   each item in the list is a complex value
+                //              |     | (if 4th bit is 1 is map)
+                //              |     |   remaining bits are the number of key_index and 
+                //              |     |   complex_value pairs to follow (same as properties)
```

Comment:
Can we simply make these `list -> 8`, `map > 9`? The extra bit is confusing and doesn't buy us anything, because we already have 9 values (0-8) for the Id anyway.